test/try-snapshot.js

```diff
@@ -0,0 +1,94 @@
+'use strict';
+require('../lib/chalk').set();
+require('../lib/worker/options').set({color: false});
+
+const path = require('path');
+const {test} = require('tap');
+const snapshotManager = require('../lib/snapshot-manager');
+const Test = require('../lib/test');
+const ContextRef = require('../lib/context-ref');
+
+function setup(title, manager, fn) {
+	return new Test({
+		fn,
+		failWithoutAssertions: true,
+		metadata: {type: 'test', callback: false},
+		contextRef: new ContextRef(),
+		registerUniqueTitle: () => true,
+		title,
+		compareTestSnapshot: options => manager.compare(options)
+	});
+}
+
+test(async t => {
+	// Set to `true` to update the snapshot, then run:
+	// "$(npm bin)"/tap --no-cov -R spec test/try-snapshot.js
+	//
+	// Ignore errors and make sure not to run tests with the `-b` (bail) option.
+	const updating = false;
+
+	const projectDir = path.join(__dirname, 'fixture');
+	const manager = snapshotManager.load({
+		file: path.join(projectDir, 'try-snapshot.js'),
+		projectDir,
+		fixedLocation: null,
+		updating,
+		recordNewSnapshots: updating
+	});
+
+	await t.test('try-commit snapshots serially', t => {
+		const ava = setup('serial', manager, a => {
+			a.snapshot('hello');
+
+			const attempt1 = t2 => {
+				t2.snapshot(true);
+				t2.snapshot({boo: 'far'});
+			};
+
+			const attempt2 = t2 => {
+				t2.snapshot({foo: 'bar'});
+			};
+
+			return a.try(attempt1).then(first => {
```

Comment:
It may be clearer to use `const attempt1 = try(t2 => …)` rather than setting up the functions. It took me a while to realize they weren't executing yet.